### PR TITLE
Documentation of NuGet package rules

### PR DESCRIPTION
**Pull request by Corniel**
Useful overview on all NuGet package related rules. See also: https://github.com/dotnet-project-file-analyzers/dotnet-project-file-analyzers/issues/26

### CODE REVIEW

File: general/nuget-packages.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# NuGet packages
+microsoft provides [best practices](https://learn.microsoft.com/nuget/create-packages/package-authoring-best-practices)
+to give NuGet package authors a reference to create and publish high-quality
+packages.
+
+.NET project file analyzers provided rules for most of them:
+
+| Rule                             | Recomandation
```
Comment (Gemberkoekje) [inline general/nuget-packages.md:8]: Recommendation

File: rules/Proj0200.md [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-be packed when the `dotnet pack` command
-is executed
-([more info](https://learn.microsoft.com/en-us/nuget/create-packages/creating-a-package-msbuild)).
-Projects that have `IsPackable` set to `false`
-will not be packed when the `dotnet pack`
-command is executed.
+To prevent confusion, explicitly define the `<IsPackable>` node. Setting the 
+`IsPackable` to `true` indicates that the package should be packed when the
+`dotnet pack` [command is executed]((https://learn.microsoft.com/nuget/create-packages/creating-a-package-msbuild)).
+This result in the creation of a [NuGet package](../general/nuget-packages.md).
```
Comment (Gemberkoekje) [inline rules/Proj0200.md:5]: results

File: rules/Proj0200.md [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
+
+Projects that have `IsPackable` set to `false` will not be packed when the
+`dotnet pack` command is executed. This is particular useful as it prevents the
```
Comment (Gemberkoekje) [inline rules/Proj0200.md:8]: particularly

**Review by Gemberkoekje — APPROVED** [review]
Couple of small typos

File: rules/Proj0212.md [hunk 4]
```diff
@@ -1,17 +1,14 @@
...
 `<IsPackable>` node with value `false`.
 
-While in principle only defining `<PackageIcon>`
-should be enough. Not all external tools are capable
-of correctly handling this modern type of package icon
-definition. Similarly, while `<PackageIconUrl>` is
-deprecated, some older tools can not correctly handle
-the modern option. For maximum compatibility it is
-therefore recommended to define both.
+While in principle only defining `<PackageIcon>` should be enough. Not all
```
Comment (LauraXiulan) [inline rules/Proj0212.md:7]:
```suggestion
While in principle only defining `<PackageIcon>` should be enough, not all
```

File: rules/Proj0200.md (hunk 2 above)
Comment (Corniel) [inline rules/Proj0200.md:5]:
```suggestion
This results in the creation of a [NuGet package](../general/nuget-packages.md).
```

File: rules/Proj0200.md [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
-Projects that have `IsPackable` set to `false`
-will not be packed when the `dotnet pack`
-command is executed.
+To prevent confusion, explicitly define the `<IsPackable>` node. Setting the 
+`IsPackable` to `true` indicates that the package should be packed when the
+`dotnet pack` [command is executed]((https://learn.microsoft.com/nuget/create-packages/creating-a-package-msbuild)).
+This results in the creation of a [NuGet package](../general/nuget-packages.md).
+
+Projects that have `IsPackable` set to `false` will not be packed when the
+`dotnet pack` command is executed. This is particular useful as it prevents the
```
Comment (Corniel) [inline rules/Proj0200.md:8]:
```suggestion
`dotnet pack` command is executed. This is particularly useful as it prevents the
```

**Review by LauraXiulan — APPROVED** [review]

**Pull request merged**